pset/doc.go

```diff
@@ -0,0 +1,353 @@
+// It consists of a collection of methods that, basically, has the purpose of adding any new field to an existing partial
+// input (included issuance or reissuance placed in the unsigned tx) or output.
+// It also allows to add new inputs or outputs to the underlying unsigned transaction.
+// The updater can be instantiated by calling the NewUpdater factory function passing a partial transasction object.
+func ROLE_2_Updater() *PSET_PACKAGE { return nil }
+
+// At the moment the blinder role is designed to blind ALL the outputs of the partial transaction but
+// this will change soon, letting one to blind only the set of outputs he wants.
+// Also, this version of the blinder requires that all the private keys necessary
+// to unblind all the confidential inputs used must be provided.
+// Given this, the pset package is not useful in case multiple parties want to
+// create a transaction by joining their inputs/outputs since they would need to
+// reveal their blinding private keys and share them with the one encharged of assuming the blinder role.
+// The pset package will change in the future to support the use case mentioned before, but this is not yet planned in the development.
+func ROLE_3_Blinder() *PSET_PACKAGE { return nil }
+
+// The signer is in charge of checking that when adding a signature to an input of the pset,
+// this is valid and that also the pset is correctly structured. Given that, this role is implemented
+// as a function Sign of the *Updater type. This function accepts an input index, a signature, a public key,
+// and one between a redeem or witness script and checks that the signature is valid against the given script and pubkey,
+// along with setting the partial input's signature script to the one provided.
+func ROLE_4_Signer() *PSET_PACKAGE { return nil }
+
+// The finalizer takes a partial transaction and combines every input's PartialSignature into the final input's
+// SignatureScript. After finalizing, the partial transaction is complete, and it's ready to be extracted from the
+// Pset wrapper and broadcasted to the network. This role is accomplished by a Finalize function that accepts a *Pset
+//instance and an input index, and performs the operations described above, returning an error if any occurs during the process.
+//It previously checks that the provided partial transaction is valid in the sense that it's ready to be finalized; otherwise,
+// an error is returned. A handy FinalizeAll that runs the above method for every input of the provided *Pset is also exported.
+func ROLE_5_Finalizer() *PSET_PACKAGE { return nil }
+
+// The extractor is a simple Extract function expecting a finalized partial transaction that returns the final signed
+// transaction by adding the signatures of the partial inputs to the underlying unsigned transaction.
+func ROLE_6_Extractor() *PSET_PACKAGE { return nil }
+
+/*
+This is a simple exemplification on how to perform a basic P2PKH transaction using the PSET package
```

Comment:
I woudl put this as Example dropdown https://godoc.org/github.com/fluhus/godoc-tricks#pkg-examples